app/geo/gui/src/ts/twxplore/gui/geo/components/filterSliders/FilterSliders.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from "react";
+import {makeStyles, Theme, createStyles} from "@material-ui/core/styles";
+import Typography from "@material-ui/core/Typography";
+import Slider from "@material-ui/core/Slider";
+import {connect, useSelector} from "react-redux";
+import {MapState} from "../../states/map/MapState";
+import {RootState} from "../../states/root/RootState";
+import {setFilter} from "kepler.gl/actions";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      width: 300,
+    },
+    margin: {
+      height: theme.spacing(3),
+    },
+  })
+);
+
+function valuetext(value: number) {
+  return `${value}`;
+}
+
+const FilterSlidersImpl: React.FunctionComponent<{type: string}> = ({type}) => {
+  const classes = useStyles();
+  const state: MapState = useSelector(
+    (rootState: RootState) => rootState.app.map
+  );
+
+  const typeRange = state.typesRanges[type];
+
+  const handleChange = (event: any, newValue: number | number[]) => {
+    console.log(type);
+    console.log(newValue);
+    setFilter(0, type, newValue);
+  };
+
+  return (
+    <div className={classes.root}>
+      <div className={classes.margin} />
+      {Object.keys(typeRange).map(attribute => {
+        //attribute being an attribute of a feature e.g. timestamp, frequency
+        const attributeProperties = (typeRange as any)[attribute]; //e.g. timestamp:{min,max}, frequency:{min, max}
+        if (attributeProperties) {
```

Comment:
If you have an if else that's unbalanced like this, do 

`
if (!attributeProperties) { return <div></div>; }
`

Then you don't need both branches, and can save an indentation.